tests/test_resolver.py

```diff
@@ -101,6 +101,20 @@ def test_non_assetized_child_ref_assetized_grandchild():
     assert_parking_lot_structure(stage)
 
 
+# Will attempt to resolve `bal:///` (i.e. no path component), which
+# will trigger an exception in BAL internals.
+def test_error_triggering_asset_ref(capfd):
+    open_stage(
+        "resources/integration_test_data/error_triggering_asset_ref/parking_lot.usd"
+    )
+
+    logs = capfd.readouterr()
+    assert (
+        "OpenAssetIO error in UsdOpenAssetIOResolver::_GetExtension: RuntimeError: 130:"
+        " Malformed BAL reference: Missing entity name in path component" in logs.err
```

Comment:
[consider] Is there a middle ground here where we're not coupling to BALs specific messages? Just we've had this  a few times before where we create a lot of maintenance churn, say when BAL changes its error messages.

Reply:
How about leaving the `RuntimeError: 130`, but removing the actual message?  It's still somewhat tied to BAL, but then, the whole test relies on knowledge of BAL's quirks to trigger a specific failure mode.

Reply:
Removed the message part in b6eb77d0

Reply:
Nice one, like it 👍